webapp/channels/src/components/suggestion/suggestion_list.tsx

```diff
@@ -13,50 +12,62 @@ import {isEmptyObject} from 'utils/utils';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 import LoadingSpinner from 'components/widgets/loading/loading_spinner';
 
-// When this file is migrated to TypeScript, type definitions for its props already exist in ./suggestion_list.d.ts.
-
-export default class SuggestionList extends React.PureComponent {
-    static propTypes = {
-        ariaLiveRef: PropTypes.object,
-        inputRef: PropTypes.object,
-        open: PropTypes.bool.isRequired,
-        position: PropTypes.oneOf(['top', 'bottom']),
-        renderDividers: PropTypes.arrayOf(PropTypes.string),
-        renderNoResults: PropTypes.bool,
-        onCompleteWord: PropTypes.func.isRequired,
-        preventClose: PropTypes.func,
-        onItemHover: PropTypes.func.isRequired,
-        pretext: PropTypes.string.isRequired,
-        cleared: PropTypes.bool.isRequired,
-        matchedPretext: PropTypes.array.isRequired,
-        items: PropTypes.array.isRequired,
-        terms: PropTypes.array.isRequired,
-        selection: PropTypes.string.isRequired,
-        components: PropTypes.array.isRequired,
-        suggestionBoxAlgn: PropTypes.object,
+interface Props {
+    ariaLiveRef?: React.Ref<HTMLDivElement>;
```

Comment:
Would you mind changing the type of `ariaLiveRef` and `inputRef` to be `React.RefObject` instead? I think that should fix the complicated typing we discussed [here](https://github.com/mattermost/mattermost/pull/23837#discussion_r1245456660).

If that doesn't fix it right away though, feel free to leave this as-is though. TS is nicer than plain JS in a lot of cases, but it's definitely possible to spend a waste a bunch of time trying to get things "correct"